docs/reference/esql/processing-commands/limit.asciidoc

```diff
@@ -17,11 +17,27 @@ The maximum number of rows to return.
 *Description*
 
 The `LIMIT` processing command enables you to limit the number of rows that are
-returned. If not specified, `LIMIT` defaults to `500`.
+returned.
+// tag::limitation[]
+Queries do not return more than 10,000 rows, regardless of the `LIMIT` command's
+value.
 
-A query does not return more than 10,000 rows, regardless of the `LIMIT` value.
-You can change this with the `esql.query.result_truncation_max_size` static
-cluster setting.
+This limit only applies to the number of rows that are retrieved by the query
+and displayed in Discover. Queries and aggregations run on the full data set.
```

Comment:
This does not only apply to discover. When using ESQL via curl, the same Max limit applies.

Can also be explained by saying that execution behaves as if there always was a LIMIT appended at the end.

Reply:
I've removed the "and displayed in Discover" in https://github.com/elastic/elasticsearch/pull/101831/commits/bf5a1d2f36f3932f3b4d81a55b88faa458e09345